File: Assets/Disease Destroyer Assets/Scripts/whiteCellMovement.js
var splat : AudioSource;//death sound
var oneShotAudio=false;//singleton for death sound
private var maxX = 480;//bounding box
private var minX = -480;
private var maxY = 480;
private var minY = -480;
var speed  = 85;//movement multiplier
private var health=3;
private var emitter: ParticleEmitter;
public var tex1: Texture2D;
public var tex2: Texture2D;
public var tex3: Texture2D;

private var tChange: float = 0; // force new direction in the first Update
private var startTime;
// Use this for initialization

function Start () {
	emitter = GetComponentInChildren(ParticleEmitter);   
    if (emitter) {
        emitter.emit = false;
    }
    this.startTime=Time.time;
}


function FixedUpdate () {
	if((Time.time>(this.startTime+1))){//only move if the game isn't over
    	if(Time.time>tChange){
    		attack();
    		print("attacking");
    		tChange=Time.time+5;//wait the 2 seconds for the cell to decay before choosing a new descision
    	}
    }
    else{
    	rigidbody.position=Vector3.zero;
		rigidbody.velocity=Vector3.zero;
    }
    if(rigidbody.transform.position.x>maxX){//if the virus leaves the bounding box bring it back in and send it toward the center
    	rigidbody.transform.position.x=maxX;
    	rigidbody.velocity.x=-Mathf.Abs(rigidbody.velocity.x*.5);
    }
    if(rigidbody.transform.position.x<minX){
    	rigidbody.transform.position.x=minX;
    	rigidbody.velocity.x=Mathf.Abs(rigidbody.velocity.x*.5);
    }
    if(rigidbody.transform.position.y>maxY){
    	rigidbody.transform.position.y=maxY;
    	rigidbody.velocity.y=-Mathf.Abs(rigidbody.velocity.y*.5);
    }
    if(rigidbody.transform.position.y<minY){
    	rigidbody.transform.position.y=minY;
    	rigidbody.velocity.y=Mathf.Abs(rigidbody.velocity.y*.5);
    }

		rigidbody.velocity=rigidbody.velocity*1.01;
	
}
function OnCollisionEnter(collision : Collision) {
	//var tex = Resources.Load("cell3");
	
	if(collision.gameObject.name.Contains("Border")){//bounce off borders
		if(rigidbody.position.x>0&&Mathf.Abs(rigidbody.position.x)>450){
			rigidbody.velocity.x=-Mathf.Abs(rigidbody.velocity.x);
		}
		if(rigidbody.position.x<0&&Mathf.Abs(rigidbody.position.x)>450){
			rigidbody.velocity.x=Mathf.Abs(rigidbody.velocity.x);
		}
		if(rigidbody.position.y>0&&Mathf.Abs(rigidbody.position.y)>450){
			rigidbody.velocity.y=-Mathf.Abs(rigidbody.velocity.y);
		}
		if(rigidbody.position.y<0&&Mathf.Abs(rigidbody.position.y)>450){
			rigidbody.velocity.y=Mathf.Abs(rigidbody.velocity.y);
		}
    }
    if(this.startTime==null){
		this.startTime=Time.time;
	}/*
	if(collision.gameObject.name.Contains("Character")){
		renderer.material.mainTexture = tex3;
	}*/
	if(collision.gameObject.name=="Character"){
    	GameObject.FindGameObjectWithTag("Player").GetComponent(playerMovement).startSlow();
    }
   

    if(collision.gameObject.name.Contains("bullet")){
    	health=health-1;
		if(health==2){renderer.material.mainTexture = tex2;}
		if(health==1){renderer.material.mainTexture= tex3;}
		if(health==0){
    		if(!oneShotAudio){
				splat.Play();
				oneShotAudio=true;
			}
    		Destroy (gameObject);
			Destroy (this);
			Destroy (rigidbody);
		}
    }
}
function attack(){
	var player=GameObject.FindGameObjectWithTag("Player");
	var position = transform.position; //find which direction the player is
	if(player!=null){//head randomly in the players general direction
    	if(player.transform.position.y<position.y){
    		yDir=Random.Range(-2.0,-1.0);
    	}
    	else if(player.transform.position.y>position.y){
    		yDir=Random.Range(1.0,2.0);
    	}
    	if(player.transform.position.x<position.x){
    		xDir=Random.Range(-2.0,-1.0);
    		
    	}
   		else if(player.transform.position.x>position.x){
    		xDir=Random.Range(1.0,2.0);
   	 	}
   	 	rigidbody.velocity=rigidbody.velocity+Vector3(xDir*speed,yDir*speed,0);
    }
}
function kill(){
	health=health-1;
		if(health==2){renderer.material.mainTexture = tex2;}
		if(health==1){renderer.material.mainTexture= tex3;}
		if(health==0){
			if(emitter!=null){
				emitter.emit=true;
				emitter.transform.parent=null; // detach particle system
				Destroy(emitter.gameObject, 3);
			}
			if(!oneShotAudio){//play death sound and destroy
				splat.Play();
				oneShotAudio=true;
			}
    		Destroy (gameObject);
			Destroy (this);
			Destroy (rigidbody);
		}
}

File: Assets/Disease Destroyer Assets/Scripts/whiteCellMovement.cs
using UnityEngine;

[RequireComponent(typeof(Rigidbody),typeof(Renderer))]
public class whiteCellMovement : MonoBehaviour
{
    public AudioSource splat; public bool oneShotAudio; public float speed=85; public Texture2D tex1,tex2,tex3; public ParticleSystem emitter;
    int health=3;float tChange,startTime;Rigidbody body;Renderer visual;
    void Awake(){body=GetComponent<Rigidbody>();visual=GetComponent<Renderer>();emitter=GetComponentInChildren<ParticleSystem>();}
    void Start(){startTime=Time.time;if(emitter)emitter.Stop();}
    void FixedUpdate(){if(Time.time>startTime+1&&Time.time>tChange){Attack();tChange=Time.time+5;}else if(Time.time<=startTime+1){body.position=Vector3.zero;body.linearVelocity=Vector3.zero;}body.linearVelocity*=1.01f;Bounce();}
    void Attack(){var p=GameObject.FindGameObjectWithTag("Player");if(!p)return;body.linearVelocity+=new Vector3(p.transform.position.x<transform.position.x?Random.Range(-2f,-1f)*speed:Random.Range(1f,2f)*speed,p.transform.position.y<transform.position.y?Random.Range(-2f,-1f)*speed:Random.Range(1f,2f)*speed,0);}
    void OnCollisionEnter(Collision c){if(c.gameObject.name=="Character")c.gameObject.GetComponent<playerMovement>()?.startSlow();if(c.gameObject.name.Contains("bullet"))kill();}
    public void kill(){health--;if(visual){if(health==2&&tex2)visual.material.mainTexture=tex2;if(health==1&&tex3)visual.material.mainTexture=tex3;}if(health<=0){if(emitter){emitter.transform.SetParent(null);emitter.Play();Destroy(emitter.gameObject,3);}if(splat&&!oneShotAudio){splat.Play();oneShotAudio=true;}Destroy(gameObject);}}
    void Bounce(){var p=body.position;var v=body.linearVelocity;if(p.x>480){p.x=480;v.x=-Mathf.Abs(v.x*.5f);}if(p.x<-480){p.x=-480;v.x=Mathf.Abs(v.x*.5f);}if(p.y>480){p.y=480;v.y=-Mathf.Abs(v.y*.5f);}if(p.y<-480){p.y=-480;v.y=Mathf.Abs(v.y*.5f);}body.position=p;body.linearVelocity=v;}
}
